app/models/server/models/_BaseDb.js

```diff
@@ -10,11 +10,13 @@ const baseName = 'rocketchat_';
 
 const trash = new Mongo.Collection(`${ baseName }_trash`);
 try {
-	trash._ensureIndex({ collection: 1 });
+	trash._ensureIndex({ __collection__: 1 });
 	trash._ensureIndex(
 		{ _deletedAt: 1 },
 		{ expireAfterSeconds: 60 * 60 * 24 * 30 },
 	);
+
+	trash._ensureIndex({ __collection__: 1, _deletedAt: 1, rid: 1 });
```

Comment:
I didn't find a query where `rid` is used along with `__collection__` and `_deletedAt`.. I think the best index would be without `rid`

Reply:
It's used [here](https://github.com/RocketChat/Rocket.Chat/blob/b7419757d65097742951a6ddaf37b272259c56da/server/publications/room/emitter.js#L9) calling the `trashFind` which receives a query, we found this trough database profiling

Reply:
oh.. ok, that's hard to track 😞 

but I've [found recently](https://docs.mongodb.com/manual/core/index-compound/#compound-index-prefix) that field order is important to compound indexes.. so in this case, I think the proper order would be:
```suggestion
	trash._ensureIndex({ rid: 1, __collection__: 1, _deletedAt: 1 });
```

so even this index will be used if do not include `_deletedAt_` on your query.

Reply:
Makes sense